Mobile/ShopLens/Droid/Activities/MainActivity.cs

```diff
@@ -90,73 +107,165 @@ protected override void OnCreate(Bundle savedInstanceState)
+        }
+
+        private void RunUserTutorial()
+        {
+            var message = ConfigurationManager.AppSettings["InitialTutorialMsg"];
+            tts.Speak(message, QueueMode.Flush, null, needUserAnswerId);
+        }
+
+        private void ContinueUserTutorial()
+        {
+            var message = ConfigurationManager.AppSettings["FollowUpTutorialMsg"];
+            tts.Speak(message, QueueMode.Flush, null, needUserAnswerId);
+        }
+
+        // TTS Engine method called when TTS is initialized.
+        public void OnInit([GeneratedEnum] OperationResult status)
+        {
+            if (status == OperationResult.Error)
+            {
+                tts.SetLanguage(Locale.Default);
+            }
+            else if (status == OperationResult.Success)
+            {
+                tts.SetLanguage(Locale.Us);
+            }
+
+            if (tutorialNeeded)
+            {
+                RunUserTutorial();
+            }
+            else
+            {
+                var welcomeMsg = ConfigurationManager.AppSettings["WelcomeBackMsg"];
+                tts.Speak(welcomeMsg, QueueMode.Flush, null, needUserAnswerId);
+            }
+        }
+
+        private void TtsStoppedSpeaking(object sender, UtteranceIdArgs e)
+        {
+            if (e.UtteranceId == needUserAnswerId)
+            {
+                voiceRecognizer.RecognizePhrase(this);
+            }
+        }
+
+        private void OnVoiceRecognitionResults(object sender, ShopLensSpeechRecognizedEventArgs e)
+        {
+            var results = e.Phrase;
+
+            if (results == ConfigurationManager.AppSettings["CmdOpenCamera"])
+            {
+                StartCameraIntent();
+            }
+            else if (results == ConfigurationManager.AppSettings["CmdOpenCart"])
+            {
+                StartCartIntent();
+            }
+            else if (results == ConfigurationManager.AppSettings["CmdOpenList"])
+            {
+                StartListIntent();
+            }
+            else if (results == ConfigurationManager.AppSettings["CmdHelp"])
+            {
+                var helpMessage = ConfigurationManager.AppSettings["MainHelpMsg"];
+                tts.Speak(helpMessage, QueueMode.Flush, null, needUserAnswerId);
+            }
+            else if (results == ConfigurationManager.AppSettings["CmdTutorialLikeShopLens"] && tutorialNeeded)
+            {
+                ContinueUserTutorial();
+            }
+            else
+            {
+                tts.Speak(askUserToRepeat, QueueMode.Flush, null, needUserAnswerId);
+            }
+        }
+
+        private void StartCameraIntent()
+        {
+            var intentCam = new Intent(this, typeof(CameraActivity));
+            StartActivity(intentCam);
+        }
+
+        private void StartCartIntent()
+        {
+            var intentCart = new Intent(this, typeof(ShoppingCartActivity));
+            intentCart.PutExtra(noTalkBackIntentKey, noTalkBack);
```

Comment:
That does make sense, but actually its bad practice to do that. Take a look: https://stackoverflow.com/questions/30058835/static-fields-in-android

Reply:
Sure, it's a bad practice to pass around data through static variables, but this is a flag if I understand correctly? So it's just a yes/no configuration parameter, which is the same for the whole program.

If not static, maybe just pull it from the prefs (since they cache the value anyway IIRC)